### PR TITLE
Ladder week implementation

### PR DESCRIPTION
**Pull request by louvenarde**
Those changes add a tab for the Ladder week, as @Rowey plans to launch another one : and maybe people will launch other ladder weeks after him aswell.

image

There is still debate over the way the configuration files for the Ladderweek should be stored.
Apart from this point, everything is done.
Ladder week ruleset is here : https://forums.faforever.com/viewtopic.php?f=26&t=15474&sid=91128f75bf78bb7e5477e9314ee95f9b

Could you please review my code and tell me what you think ? I would be glad to clean it / improve it while waiting for solutions concerning the configuration file.

### CODE REVIEW

File: public/js/app/ladderWeek.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    
+    // Score
+    var points = document.createElement("td");
+    tr.appendChild(points);
+    points.innerHTML = player.points;
+    
+    // W/L/D
+    var wld = document.createElement("td");
+    tr.appendChild(wld);
+    wld.innerHTML = player.wld.w+"/"+player.wld.l+"/"+player.wld.d;
```
Comment (Geosearchef) [inline public/js/app/ladderWeek.js]: is there no better way to format text in JS?

File: public/js/app/ladderWeek.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    
+    
+    // Time played
+    var timeplayed = document.createElement("td");
+    tr.appendChild(timeplayed);
+    timeplayed.innerHTML = (moment(player.secondsPlayed*1000).hours()-1)+"h "+moment(player.secondsPlayed*1000).minutes()+"m "+moment(player.secondsPlayed*1000).seconds()+"s";
+  }
```
Comment (Geosearchef) [inline public/js/app/ladderWeek.js]: extract method for createElement, appendChild and set innerHTML

File: public/js/app/ladderWeek.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    // W/L/D
+    var wld = document.createElement("td");
+    tr.appendChild(wld);
+    wld.innerHTML = player.wld.w+"/"+player.wld.l+"/"+player.wld.d;
+    
+    
+    // Time played
+    var timeplayed = document.createElement("td");
+    tr.appendChild(timeplayed);
+    timeplayed.innerHTML = (moment(player.secondsPlayed*1000).hours()-1)+"h "+moment(player.secondsPlayed*1000).minutes()+"m "+moment(player.secondsPlayed*1000).seconds()+"s";
```
Comment (Geosearchef) [inline public/js/app/ladderWeek.js]: extract method for time generation pls, this is not very readable

File: public/js/app/ladderWeek.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  $("#navigation-previous-"+category).addClass("disabled");
+  $("#navigation-next-"+category).addClass("disabled");
+  $("#navigation-last-"+category).addClass("disabled");
+  if (currentPage[category] < lastPage[category]){
+      $("#navigation-next-"+category).removeClass("disabled");
+      $("#navigation-last-"+category).removeClass("disabled");
+  }
+  if (currentPage[category] > 1){
+      $("#navigation-first-"+category).removeClass("disabled");
+      $("#navigation-previous-"+category).removeClass("disabled");
```
Comment (Geosearchef) [inline public/js/app/ladderWeek.js]: extract method for

`$("#navigation-XXXXXXX-"+category).removeClass("disabled");`

File: public/js/app/ladderWeek.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+  }
+  const elementId = "players"+category;
```
Comment (Geosearchef) [inline public/js/app/ladderWeek.js]: why introduce variable?

File: public/js/app/ladderWeek.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    $("#navigation-first-"+catName).click( function() {
+      currentPage[catName] = 1;
+      getPage(catName, currentPage[catName], pageSize);
+    });
+
+    $("#navigation-last-"+catName).click( function() {
+      currentPage[catName] = lastPage[catName];
+      getPage(catName, currentPage[catName], pageSize);
+    });
+}
```
Comment (Geosearchef) [inline public/js/app/ladderWeek.js]: this for block has a lot of duplicate code

File: public/js/app/ladderWeek.js (hunk 5 above)
Comment (louvenarde) [reply public/js/app/ladderWeek.js]: easier debugging 😛

File: public/js/app/ladderWeek.js (hunk 6 above)
Comment (louvenarde) [reply public/js/app/ladderWeek.js]: it is original though, from the leaderboards.js. Will fix.

**Comment by Brutus5000** [comment]
Due to lack of maintainers who can give their final ok I merge this. Should we come up with problems on next server update, we need to rollback.

**Pull request merged**